pkg/ecsdeploy/ecs_deploy.go

```diff
@@ -0,0 +1,102 @@
+package ecsdeploy
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"go.uber.org/zap"
+)
+
+// ECSClusterServiceDeployer defines an ECS Service configuration on a particular ECS Cluster.
+type ECSClusterServiceDeployer struct {
+	ECSCluster string
+	ECSService string
+	DryRun     bool
+	Logger     *zap.Logger
+	ECSClient  *ecs.ECS
+}
+
+func stringToStringSlice(originalString string) []string {
+	slice := make([]string, 1)
+	slice[0] = originalString
+	return slice
+}
+
+// GetServiceTaskDefiniton returns the primary/active task definition arn for the service/cluster combination
+func (e *ECSClusterServiceDeployer) GetServiceTaskDefiniton() (*ecs.TaskDefinition, error) {
+	describeServiceInput := &ecs.DescribeServicesInput{
+		Cluster:  aws.String(e.ECSCluster),
+		Services: aws.StringSlice(stringToStringSlice(e.ECSService)),
+	}
+
+	response, err := e.ECSClient.DescribeServices(describeServiceInput)
+	taskDefinitionArn := response.Services[0].TaskDefinition
+
+	if err != nil {
+		return nil, err
+	}
+
+	e.Logger.Info("Found task definition arn.", zap.String("taskDefinitionArn", *taskDefinitionArn))
+
+	describeTaskInput := &ecs.DescribeTaskDefinitionInput{
+		TaskDefinition: taskDefinitionArn,
+	}
+
+	r, err := e.ECSClient.DescribeTaskDefinition(describeTaskInput)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return r.TaskDefinition, nil
+}
+
+// RegisterUpdatedTaskDefinition registers a new task definition based on the existing service definition with updated container images and returns the new taskdefinition arn and errors
+func (e *ECSClusterServiceDeployer) RegisterUpdatedTaskDefinition(taskDefinition *ecs.TaskDefinition, containerMap map[string]map[string]string) (*ecs.TaskDefinition, error) {
+	for containerName := range containerMap {
+		for idx, containerDefinition := range taskDefinition.ContainerDefinitions {
+			containerDefName := *containerDefinition.Name
+			if containerDefName == containerName {
+				*taskDefinition.ContainerDefinitions[idx].Image = containerMap[containerName]["image"]
+			}
```

Comment:
If I'm understanding this correctly, if there is no matching containerDefName, the taskDefinition.ContainerDefinitions will remain unchanged, but the task definition will still be registered on line 76 below? Would it be possible for there to be other containerDefNames, or is the intention to redeploy regardless?

Reply:
Hrm -- I think if there's no matching containerDefName, will there even be a specification for the ContainerDefinition? What happens if there's an empty one? I feel like that will probably break, won't it?

Reply:
I cleaned this up a bit and will push the change in a few minutes.... 
@amitch23 confirming the logic you stated. if there is no matching containerDefName the taskDefinition.ContainerDefinitions will remain unchanged and will be registered with the new task definition. The intention is to redeploy regardless.

@cblkwell I think if there is an empty ContainerDefinition, it will probably break. However, that shouldn't happen since we are modifying an existing Container Definition.  The task definition we submit will just be a duplicate of the existing definition.